### PR TITLE
Add case list filters to case summary

### PR DESCRIPTION
**Pull request by orangejenny**
##### SUMMARY
https://trello.com/c/a3tQZTOu/118-add-case-list-filters-to-app-case-summary

##### RISK ASSESSMENT / QA PLAN
Minor, no QA.

##### PRODUCT DESCRIPTION
I added these to the list of case properties, displaying them the same way as calculated case proeprties and marking them with `[Filter]`:

Screen Shot 2020-07-27 at 2 38 01 PM

### CODE REVIEW

File: corehq/apps/app_manager/app_schemas/app_case_metadata.py [hunk 1]
```diff
@@ -513,8 +519,8 @@ def add_property_error(self, case_type, case_property, form_id, message):
             form.errors.append(message)
         return prop
 
-    def add_property_detail(self, detail_type, root_case_type, module_id, column):
-        field = column.field
+    def add_property_detail(self, detail_type, root_case_type, module_id,
+                            field, header, is_detail_calculation=False):
```
Comment (proteusvacuum) [inline corehq/apps/app_manager/app_schemas/app_case_metadata.py:523]: nit: I don't love the addition of a boolean kwarg. Not sure if you can make a new method `.add_property_detail_calculation` instead?

File: corehq/apps/app_manager/app_schemas/app_case_metadata.py [hunk 2]
```diff
@@ -57,9 +57,15 @@ def _add_module_contributions(self):
...
         for column in module.case_details.long.columns:
-            self.meta.add_property_detail('long', module.case_type, module.unique_id, column)
+            self.meta.add_property_detail('long', module.case_type, module.unique_id,
+                                          column.field, column.header, column.useXpathExpression)
         for column in module.case_details.short.columns:
-            self.meta.add_property_detail('short', module.case_type, module.unique_id, column)
+            self.meta.add_property_detail('short', module.case_type, module.unique_id,
+                                          column.field, column.header, column.useXpathExpression)
+        if module.case_details.short.filter:
+            header = {self.app.default_language: _("[Filter]")}     # Fake a localized "header"
```
Comment (proteusvacuum) [inline corehq/apps/app_manager/app_schemas/app_case_metadata.py:66]: What happens if you are looking at a different language than the default? Does it just pick this one?

**Review by proteusvacuum — APPROVED** [review]
Looks fine to me. I remember now that it was hard to match up calculations with the actual properties that are used in those calculations. (so in your example, we'd actually want a single row for `rating` which would include the fact that it was used as a calculation and a filter in different case lists). Probably out of scope, and you can somewhat get around it by using the search box.

File: corehq/apps/app_manager/app_schemas/app_case_metadata.py (hunk 1 above)
Comment (orangejenny) [reply corehq/apps/app_manager/app_schemas/app_case_metadata.py:523]: Going to leave this as-is - took a look at adding a second method and think it made things more confusing.

File: corehq/apps/app_manager/app_schemas/app_case_metadata.py (hunk 2 above)
Comment (orangejenny) [reply corehq/apps/app_manager/app_schemas/app_case_metadata.py]: I'm guessing it'll prepend a label, so it'd appear like `[en] [Filter]` if the default language was English and you viewed it in another language. Which is weird. Added 55c64a41510 so it'll display the same in all languages.

**Review by proteusvacuum — APPROVED** [review]

**Comment by orangejenny** [comment]
Forgot to respond earlier - I agree it'd be better to only have one row per property, but I can see how it's a bit complicated and am calling it out of scope for this PR.

**Pull request merged**